### ISSUE TITLE
Tip Jar depends on Google Play Services

### ISSUE THREAD

**Issue by EmberHeartshine** [pyamsoft/tetherfi#205]
I am using TetherFi installed from [F-Droid](https://f-droid.org) using the [IzzyOnDroid repo](https://apt.izzysoft.de/fdroid/) and as such, my device does not have Google Play installed on it. It may be a good idea to:

1. Remove Google integration from TetherFi to make it fully open-source (at least for releases outside Google Play; a separate Play release may be a good move if you still want to monetize that way)
2. Allow the Tip Jar to use different support methods, such as through [GitHub Sponsors](/sponsors) or [Ko-fi](https://ko-fi.com/)

Doing this would also enable the maintainer of IzzyOnDroid to include a donation link that appears at the bottom of the [app listing](https://apt.izzysoft.de/fdroid/index/apk/com.pyamsoft.tetherfi) (it of course looks a lot nicer in the [F-Droid app itself](https://github.com/pyamsoft/tetherfi/assets/50666308/61e207d4-2a47-4574-95dd-cb51d48bc4e9)).

Also, once 1 is completed, TetherFi can be submitted for inclusion on the main F-Droid repo, which would get it into the hands of thousands of more users.

**Comment by pyamsoft** [comment]
I've thought about maintaining two different builds for different stores but just don't have the band width at the moment.

**Comment by EmberHeartshine** [comment]
It can be done in a fairly straightforward manner using Gradle flavors. There's a shared codebase, so version parity isn't too hard to maintain.

**Comment by EmberHeartshine** [comment]
I wouldn't close this as complete, it should be marked wontfix or left open if it's on the back-burner imho

**Comment by pyamsoft** [comment]
tl;dr: I've re-evaluated this and don't believe this will be coming - maybe ever.

What this essentially comes down to is: If I remove the tip-jar from the application, I would not be able to replace it with any kind of alternative. Google Store policy has been argued back and forth on the internet, but I take the safe paranoid approach to it: Any website your app links to or is hosted on cannot have alternative links for tips, it all must go through Google. So that means no Kofi, no Github Sponsors, no FDroid donations. If the app ever sees a cent, it has to pass through Google.

The billing code is easy enough to remove if someone wishes to fork the app into a fully-FOSS flavor. But it would be a fork and thus wouldn't be associated with me at all. And that's fine. Take the app, fork it, change the branding and the logo and stick it up on FDroid. But sadly its not something I will be pursuing at this time.

TetherFi, and all pyamsoft Android apps, will only ever be officially distributed and supported on the Google Play Store.
